migrations/versions/0413_add_platform_admin_capable.py

```diff
@@ -0,0 +1,38 @@
+"""
+
+Revision ID: 0413_add_platform_admin_capable
+Revises: 0412_service_providers
+Create Date: 2025-03-05 17:13:13.720166
+
+"""
+
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+revision = "0413_add_platform_admin_capable"
+down_revision = "0412_service_providers"
+
+
+def upgrade():
+    op.add_column("users", sa.Column("platform_admin_capable", sa.Boolean()))
+    op.execute("UPDATE users SET platform_admin_capable = platform_admin")
+    op.alter_column("users", "platform_admin_capable", nullable=False)
+
+    op.add_column("users", sa.Column("platform_admin_redemption", sa.DateTime(), nullable=True))
+    op.drop_column("users", "platform_admin")
+
+    op.alter_column("admin_actions", "service_id", existing_type=postgresql.UUID(), nullable=True)
+
+    # We can't easily drop the enum in a downgrade so just make it idempotent
+    op.execute("ALTER TYPE admin_action_types ADD VALUE IF NOT EXISTS 'elevate_platform_admin'")
+
+
+def downgrade():
+    op.add_column("users", sa.Column("platform_admin", sa.Boolean()))
+    op.execute("UPDATE users SET platform_admin = platform_admin_capable")
```

Comment:
We tend to not have hanging platform admins on productionm so I would be tempted to simply have no platform admins on downgrade.

Reply:
Yeah that seems sensible